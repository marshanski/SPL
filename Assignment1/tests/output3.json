{
    "parties": [
        {
            "name": "a",
            "mandates": 25,
            "join_policy": "L"
        },
        {
            "name": "b",
            "mandates": 21,
            "join_policy": "M"
        },
        {
            "name": "c",
            "mandates": 18,
            "join_policy": "M"
        },
        {
            "name": "d",
            "mandates": 18,
            "join_policy": "M"
        },
        {
            "name": "e",
            "mandates": 18,
            "join_policy": "M"
        },
        {
            "name": "f",
            "mandates": 14,
            "join_policy": "L"
        },
        {
            "name": "g",
            "mandates": 6,
            "join_policy": "M"
        }
    ],
    "graph": [
        [
            0,
            2,
            1,
            4,
            5,
            1,
            4
        ],
        [
            2,
            0,
            5,
            3,
            5,
            5,
            5
        ],
        [
            1,
            5,
            0,
            5,
            4,
            2,
            5
        ],
        [
            4,
            3,
            5,
            0,
            4,
            5,
            3
        ],
        [
            5,
            5,
            4,
            4,
            0,
            5,
            5
        ],
        [
            1,
            5,
            2,
            5,
            5,
            0,
            2
        ],
        [
            4,
            5,
            5,
            3,
            5,
            2,
            0
        ]
    ],
    "agents": [
        {
            "party_id": 4,
            "selection_policy": "M"
        },
        {
            "party_id": 0,
            "selection_policy": "E"
        }
    ]
}