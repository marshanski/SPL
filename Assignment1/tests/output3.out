[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 4
            },
            {
                "id": 1,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 18,
                "parties": [
                    4
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 2
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 4
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 1
            },
            {
                "from": 0,
                "to": 6,
                "weight": 4
            },
            {
                "from": 1,
                "to": 0,
                "weight": 2
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 4,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 5
            },
            {
                "from": 1,
                "to": 6,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 4,
                "weight": 4
            },
            {
                "from": 2,
                "to": 5,
                "weight": 2
            },
            {
                "from": 2,
                "to": 6,
                "weight": 5
            },
            {
                "from": 3,
                "to": 0,
                "weight": 4
            },
            {
                "from": 3,
                "to": 1,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 5
            },
            {
                "from": 3,
                "to": 4,
                "weight": 4
            },
            {
                "from": 3,
                "to": 5,
                "weight": 5
            },
            {
                "from": 3,
                "to": 6,
                "weight": 3
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 1,
                "weight": 5
            },
            {
                "from": 4,
                "to": 2,
                "weight": 4
            },
            {
                "from": 4,
                "to": 3,
                "weight": 4
            },
            {
                "from": 4,
                "to": 5,
                "weight": 5
            },
            {
                "from": 4,
                "to": 6,
                "weight": 5
            },
            {
                "from": 5,
                "to": 0,
                "weight": 1
            },
            {
                "from": 5,
                "to": 1,
                "weight": 5
            },
            {
                "from": 5,
                "to": 2,
                "weight": 2
            },
            {
                "from": 5,
                "to": 3,
                "weight": 5
            },
            {
                "from": 5,
                "to": 4,
                "weight": 5
            },
            {
                "from": 5,
                "to": 6,
                "weight": 2
            },
            {
                "from": 6,
                "to": 0,
                "weight": 4
            },
            {
                "from": 6,
                "to": 1,
                "weight": 5
            },
            {
                "from": 6,
                "to": 2,
                "weight": 5
            },
            {
                "from": 6,
                "to": 3,
                "weight": 3
            },
            {
                "from": 6,
                "to": 4,
                "weight": 5
            },
            {
                "from": 6,
                "to": 5,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 21,
                "name": "b",
                "state": "Waiting"
            },
            {
                "id": 2,
                "mandates": 18,
                "name": "c",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 18,
                "name": "d",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 18,
                "name": "e",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 14,
                "name": "f",
                "state": "Waiting"
            },
            {
                "id": 6,
                "mandates": 6,
                "name": "g",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 4
            },
            {
                "id": 1,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 18,
                "parties": [
                    4
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 2
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 4
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 1
            },
            {
                "from": 0,
                "to": 6,
                "weight": 4
            },
            {
                "from": 1,
                "to": 0,
                "weight": 2
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 4,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 5
            },
            {
                "from": 1,
                "to": 6,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 4,
                "weight": 4
            },
            {
                "from": 2,
                "to": 5,
                "weight": 2
            },
            {
                "from": 2,
                "to": 6,
                "weight": 5
            },
            {
                "from": 3,
                "to": 0,
                "weight": 4
            },
            {
                "from": 3,
                "to": 1,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 5
            },
            {
                "from": 3,
                "to": 4,
                "weight": 4
            },
            {
                "from": 3,
                "to": 5,
                "weight": 5
            },
            {
                "from": 3,
                "to": 6,
                "weight": 3
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 1,
                "weight": 5
            },
            {
                "from": 4,
                "to": 2,
                "weight": 4
            },
            {
                "from": 4,
                "to": 3,
                "weight": 4
            },
            {
                "from": 4,
                "to": 5,
                "weight": 5
            },
            {
                "from": 4,
                "to": 6,
                "weight": 5
            },
            {
                "from": 5,
                "to": 0,
                "weight": 1
            },
            {
                "from": 5,
                "to": 1,
                "weight": 5
            },
            {
                "from": 5,
                "to": 2,
                "weight": 2
            },
            {
                "from": 5,
                "to": 3,
                "weight": 5
            },
            {
                "from": 5,
                "to": 4,
                "weight": 5
            },
            {
                "from": 5,
                "to": 6,
                "weight": 2
            },
            {
                "from": 6,
                "to": 0,
                "weight": 4
            },
            {
                "from": 6,
                "to": 1,
                "weight": 5
            },
            {
                "from": 6,
                "to": 2,
                "weight": 5
            },
            {
                "from": 6,
                "to": 3,
                "weight": 3
            },
            {
                "from": 6,
                "to": 4,
                "weight": 5
            },
            {
                "from": 6,
                "to": 5,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 21,
                "name": "b",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 18,
                "name": "c",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 18,
                "name": "d",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 18,
                "name": "e",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 14,
                "name": "f",
                "state": "Waiting"
            },
            {
                "id": 6,
                "mandates": 6,
                "name": "g",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 4
            },
            {
                "id": 1,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 18,
                "parties": [
                    4
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 2
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 4
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 1
            },
            {
                "from": 0,
                "to": 6,
                "weight": 4
            },
            {
                "from": 1,
                "to": 0,
                "weight": 2
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 4,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 5
            },
            {
                "from": 1,
                "to": 6,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 4,
                "weight": 4
            },
            {
                "from": 2,
                "to": 5,
                "weight": 2
            },
            {
                "from": 2,
                "to": 6,
                "weight": 5
            },
            {
                "from": 3,
                "to": 0,
                "weight": 4
            },
            {
                "from": 3,
                "to": 1,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 5
            },
            {
                "from": 3,
                "to": 4,
                "weight": 4
            },
            {
                "from": 3,
                "to": 5,
                "weight": 5
            },
            {
                "from": 3,
                "to": 6,
                "weight": 3
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 1,
                "weight": 5
            },
            {
                "from": 4,
                "to": 2,
                "weight": 4
            },
            {
                "from": 4,
                "to": 3,
                "weight": 4
            },
            {
                "from": 4,
                "to": 5,
                "weight": 5
            },
            {
                "from": 4,
                "to": 6,
                "weight": 5
            },
            {
                "from": 5,
                "to": 0,
                "weight": 1
            },
            {
                "from": 5,
                "to": 1,
                "weight": 5
            },
            {
                "from": 5,
                "to": 2,
                "weight": 2
            },
            {
                "from": 5,
                "to": 3,
                "weight": 5
            },
            {
                "from": 5,
                "to": 4,
                "weight": 5
            },
            {
                "from": 5,
                "to": 6,
                "weight": 2
            },
            {
                "from": 6,
                "to": 0,
                "weight": 4
            },
            {
                "from": 6,
                "to": 1,
                "weight": 5
            },
            {
                "from": 6,
                "to": 2,
                "weight": 5
            },
            {
                "from": 6,
                "to": 3,
                "weight": 3
            },
            {
                "from": 6,
                "to": 4,
                "weight": 5
            },
            {
                "from": 6,
                "to": 5,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 21,
                "name": "b",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 18,
                "name": "c",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 18,
                "name": "d",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 18,
                "name": "e",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 14,
                "name": "f",
                "state": "Waiting"
            },
            {
                "id": 6,
                "mandates": 6,
                "name": "g",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 4
            },
            {
                "id": 1,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 18,
                "parties": [
                    4
                ]
            },
            {
                "mandates": 25,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 2
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 4
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 1
            },
            {
                "from": 0,
                "to": 6,
                "weight": 4
            },
            {
                "from": 1,
                "to": 0,
                "weight": 2
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 4,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 5
            },
            {
                "from": 1,
                "to": 6,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 4,
                "weight": 4
            },
            {
                "from": 2,
                "to": 5,
                "weight": 2
            },
            {
                "from": 2,
                "to": 6,
                "weight": 5
            },
            {
                "from": 3,
                "to": 0,
                "weight": 4
            },
            {
                "from": 3,
                "to": 1,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 5
            },
            {
                "from": 3,
                "to": 4,
                "weight": 4
            },
            {
                "from": 3,
                "to": 5,
                "weight": 5
            },
            {
                "from": 3,
                "to": 6,
                "weight": 3
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 1,
                "weight": 5
            },
            {
                "from": 4,
                "to": 2,
                "weight": 4
            },
            {
                "from": 4,
                "to": 3,
                "weight": 4
            },
            {
                "from": 4,
                "to": 5,
                "weight": 5
            },
            {
                "from": 4,
                "to": 6,
                "weight": 5
            },
            {
                "from": 5,
                "to": 0,
                "weight": 1
            },
            {
                "from": 5,
                "to": 1,
                "weight": 5
            },
            {
                "from": 5,
                "to": 2,
                "weight": 2
            },
            {
                "from": 5,
                "to": 3,
                "weight": 5
            },
            {
                "from": 5,
                "to": 4,
                "weight": 5
            },
            {
                "from": 5,
                "to": 6,
                "weight": 2
            },
            {
                "from": 6,
                "to": 0,
                "weight": 4
            },
            {
                "from": 6,
                "to": 1,
                "weight": 5
            },
            {
                "from": 6,
                "to": 2,
                "weight": 5
            },
            {
                "from": 6,
                "to": 3,
                "weight": 3
            },
            {
                "from": 6,
                "to": 4,
                "weight": 5
            },
            {
                "from": 6,
                "to": 5,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 21,
                "name": "b",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 18,
                "name": "c",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 18,
                "name": "d",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 18,
                "name": "e",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 14,
                "name": "f",
                "state": "Waiting"
            },
            {
                "id": 6,
                "mandates": 6,
                "name": "g",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 4
            },
            {
                "id": 1,
                "partyId": 0
            },
            {
                "id": 2,
                "partyId": 1
            },
            {
                "id": 3,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 18,
                "parties": [
                    4
                ]
            },
            {
                "mandates": 64,
                "parties": [
                    0,
                    1,
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 2
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 4
            },
            {
                "from": 0,
                "to": 4,
                "weight": 5
            },
            {
                "from": 0,
                "to": 5,
                "weight": 1
            },
            {
                "from": 0,
                "to": 6,
                "weight": 4
            },
            {
                "from": 1,
                "to": 0,
                "weight": 2
            },
            {
                "from": 1,
                "to": 2,
                "weight": 5
            },
            {
                "from": 1,
                "to": 3,
                "weight": 3
            },
            {
                "from": 1,
                "to": 4,
                "weight": 5
            },
            {
                "from": 1,
                "to": 5,
                "weight": 5
            },
            {
                "from": 1,
                "to": 6,
                "weight": 5
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 1,
                "weight": 5
            },
            {
                "from": 2,
                "to": 3,
                "weight": 5
            },
            {
                "from": 2,
                "to": 4,
                "weight": 4
            },
            {
                "from": 2,
                "to": 5,
                "weight": 2
            },
            {
                "from": 2,
                "to": 6,
                "weight": 5
            },
            {
                "from": 3,
                "to": 0,
                "weight": 4
            },
            {
                "from": 3,
                "to": 1,
                "weight": 3
            },
            {
                "from": 3,
                "to": 2,
                "weight": 5
            },
            {
                "from": 3,
                "to": 4,
                "weight": 4
            },
            {
                "from": 3,
                "to": 5,
                "weight": 5
            },
            {
                "from": 3,
                "to": 6,
                "weight": 3
            },
            {
                "from": 4,
                "to": 0,
                "weight": 5
            },
            {
                "from": 4,
                "to": 1,
                "weight": 5
            },
            {
                "from": 4,
                "to": 2,
                "weight": 4
            },
            {
                "from": 4,
                "to": 3,
                "weight": 4
            },
            {
                "from": 4,
                "to": 5,
                "weight": 5
            },
            {
                "from": 4,
                "to": 6,
                "weight": 5
            },
            {
                "from": 5,
                "to": 0,
                "weight": 1
            },
            {
                "from": 5,
                "to": 1,
                "weight": 5
            },
            {
                "from": 5,
                "to": 2,
                "weight": 2
            },
            {
                "from": 5,
                "to": 3,
                "weight": 5
            },
            {
                "from": 5,
                "to": 4,
                "weight": 5
            },
            {
                "from": 5,
                "to": 6,
                "weight": 2
            },
            {
                "from": 6,
                "to": 0,
                "weight": 4
            },
            {
                "from": 6,
                "to": 1,
                "weight": 5
            },
            {
                "from": 6,
                "to": 2,
                "weight": 5
            },
            {
                "from": 6,
                "to": 3,
                "weight": 3
            },
            {
                "from": 6,
                "to": 4,
                "weight": 5
            },
            {
                "from": 6,
                "to": 5,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 25,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 21,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 18,
                "name": "c",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 18,
                "name": "d",
                "state": "Joined"
            },
            {
                "id": 4,
                "mandates": 18,
                "name": "e",
                "state": "Joined"
            },
            {
                "id": 5,
                "mandates": 14,
                "name": "f",
                "state": "CollectingOffers"
            },
            {
                "id": 6,
                "mandates": 6,
                "name": "g",
                "state": "CollectingOffers"
            }
        ]
    }
]
